instrumentation/logger/lib/opentelemetry/instrumentation/logger/patches/logger.rb

```diff
@@ -16,25 +16,35 @@ def format_message(severity, datetime, progname, msg)
             formatted_message = super(severity, datetime, progname, msg)
             return formatted_message if skip_instrumenting?
 
-            OpenTelemetry.logger_provider.logger(
-              name: OpenTelemetry::Instrumentation::Logger::NAME,
-              version: OpenTelemetry::Instrumentation::Logger::VERSION
+            logger_provider.logger(
+              name: Instrumentation.instance.config[:name],
+              version: Instrumentation.instance.config[:version]
             ).on_emit(
               severity_text: severity,
               severity_number: severity_number(severity),
               timestamp: datetime,
-              body: msg # New Relic uses formatted_message here. This also helps us with not recording progname, because it is included in the formatted message by default. Which seems more appropriate?
+              body: msg, # New Relic uses formatted_message here. This also helps us with not recording progname, because it is included in the formatted message by default. Which seems more appropriate?
+              context: OpenTelemetry::Context.Current
             )
             formatted_message
           end
 
+          def logger_provider=(value)
+            @logger_provider = value
+          end
+
+          def skip_instrumenting=(value)
+            @skip_instrumenting = value
+          end
```

Comment:
Since these are generic setters, you can rewrite them to `attr_writer :logger_provider, :skip_instrumenting`, after the `module Logger` line.